ner_v2/detectors/numeral/number/number_detection.py

```diff
@@ -158,13 +158,15 @@ def detect_entity(self, text, **kwargs):
         for number_value_dict, original_text in zip(number_data[0], number_data[1]):
             number_value = number_value_dict[NUMBER_DETECTION_RETURN_DICT_VALUE]
             number_unit = number_value_dict[NUMBER_DETECTION_RETURN_DICT_UNIT]
-            if self.min_digit <= self._num_digits(number_value) <= self.max_digit:
-                if self.unit_type and (number_unit is None or
-                                       self.language_number_detector.units_map[number_unit].type != self.unit_type) \
-                        and not self.detect_without_unit:
-                    continue
-                validated_number.append(number_value_dict)
-                validated_number_text.append(original_text)
+            try:
+                if self.min_digit <= self._num_digits(number_value) <= self.max_digit:
+                    if self.unit_type and (number_unit is None or self.language_number_detector.units_map[
+                       number_unit].type != self.unit_type) and not self.detect_without_unit:
+                        continue
+                    validated_number.append(number_value_dict)
+                    validated_number_text.append(original_text)
+            except (OverflowError, ValueError) as error_message:
+                ner_logger.exception(f'Exception for NumberDetector: {error_message}')
 
         if not validated_number:
```

Comment:
Hey @tanaya-b, from the platform perspective, can you explain how these two snippets are different and what is the second snippet doing in case no validated numbers are found? Can you give one example of where this concatenation of original texts helps in entity detection?

Reply:
The second instance was introduced to handle voice bots -
Eg, `My pin code is two three five eight seven nine` will provide 2,3,5,8,7,9 as separate entity values none of which will pass the _minimum six digits_ criteria. The second one appends them together to give 235879 as the whole number

Reply:
Thanks, Tanaya.